### PR TITLE
WIP list

### PR DESCRIPTION
**Pull request by vks4git**
My drafts for the Symbolic list. Feel free to discuss

### CODE REVIEW

File: symbolic-base/src/ZkFold/Symbolic/Data/List.hs [hunk 1]
```diff
@@ -1,42 +1,158 @@
...
+            \vHash vRepr (Par1 s) -> V.zipWithM (\h r -> newAssigned (\p -> p h - p r * p s)) vHash vRepr
 
-head :: List context x -> x
-head xs = let (x, _) = uncons xs in x
+        decSize :: context Par1
+        decSize = fromCircuitF lSize (\(Par1 s) -> Par1 <$> newAssigned (\p -> p s - one))
 
-tail :: List context x -> List context x
-tail xs = let (_, xs') = uncons xs in xs'
+{-- TODO: Foldable relies on Haskell types such as Bool (i.e. null = foldr (\_ _ -> False) True)
```
Comment (TurtlePU) [inline symbolic-base/src/ZkFold/Symbolic/Data/List.hs:144]: I think it shouldn't be Haskell's `Foldable`, but rather some similar typeclass which uses IVC inside

**Comment by TurtlePU** [comment]
If this compiles after fixing merge conflicts, I'd rather merge this version and then refactor it while moving in the direction I mentioned in #295. Also maybe we can abstract away the hash algorithm (and IVC device, too?) used via an additional type parameter for `List` datatype

**Comment by TurtlePU** [comment]
(Very nice PR btw)

**Review by TurtlePU — APPROVED** [review]
Nice! :shipit:

**Review by vlasin — APPROVED** [review]

**Pull request merged**